packages/ckeditor5-image/src/autoimage.js

```diff
@@ -0,0 +1,172 @@
+/**
+ * @license Copyright (c) 2003-2020, CKSource - Frederico Knabben. All rights reserved.
+ * For licensing, see LICENSE.md or https://ckeditor.com/legal/ckeditor-oss-license
+ */
+
+/**
+ * @module image/autoimage
+ */
+
+import Plugin from '@ckeditor/ckeditor5-core/src/plugin';
+import Clipboard from '@ckeditor/ckeditor5-clipboard/src/clipboard';
+import LiveRange from '@ckeditor/ckeditor5-engine/src/model/liverange';
+import LivePosition from '@ckeditor/ckeditor5-engine/src/model/liveposition';
+import Undo from '@ckeditor/ckeditor5-undo/src/undo';
+import global from '@ckeditor/ckeditor5-utils/src/dom/global';
+import { insertImage } from './image/utils';
+
+// implements pattern: http(s)://(www.)example.com/path/to/resource.ext?query=params&maybe=too
+const IMAGE_URL_REGEXP =
+	/^(http(s)?:\/\/)?[\w-]+(\.[\w-]+)+[\w._~:/?#[\]@!$&'()*+,;=%-]+\.(jpg|jpeg|png|gif|ico)\??[\w._~:/#[\]@!$&'()*+,;=%-]*$/;
```

Comment:
Some additional tests are needed as I couldn't find them:

*   image with query params
*   all supported extensions
*   special characters
*   case sensitivity: `.JPG` vs `.jpg`
*   some not-supported extensions ( like `svg`, `webp`, `mp3`, `exe` just in case :wink: )

For repetitive test cases we tend to make it in a loop ([example](https://github.com/ckeditor/ckeditor5/blob/9b6f41027da388be5baf5e317caac2b7828b94bf/packages/ckeditor5-link/tests/autolink.js#L228-L285)) - the important thing to remember is that the test name must be clear - in order to see which cases failed in the console. 

A bad example would be `It should work with URL (case 2)`. So in the linked example, I've used the full URL in the test name but you can use [other structures](https://github.com/ckeditor/ckeditor5/blob/e046735d04ee6c650501b610b6e82c6fb740a74a/packages/ckeditor5-engine/tests/dataprocessor/htmldataprocessor.js#L54-L67) as you wish.